examples/custom_criterion.cu

```diff
@@ -74,4 +74,4 @@ void throughput_bench(nvbench::state &state)
       num_values);
   });
 }
-NVBENCH_BENCH(throughput_bench).set_stopping_criterion("fixed");
+NVBENCH_BENCH(throughput_bench).set_stopping_criterion("fixed").set_throttle_threshold(0.8f);
```

Comment:
Without the changes in this PR, this code will fail to compile.

Reply:
The changes to the benchmark class are definitely necessary, but I'd rather not have this here. This example is showing how to modify the custom criterion, and the throttle settings aren't really related to that.

I'll be happy to merge this without this modification. If we want an example of this, we should add a new `throttle_detection` example with comments/annotations explaining the throttle options.

Reply:
Totally makes sense. Changes reverted.